Automattic/jetpack
# VideoPress: track upload error handler

Add error handling for .vtt track files

Fixes https://github.com/Automattic/jetpack/issues/33117

## Proposed changes:
This PR adds error handling to the .vtt track file upload process. Whenever an upload/validation error occurs, it will show an error sign (was already there, just not used for upload errors):

<img width="379" alt="image" src="https://github.com/Automattic/jetpack/assets/157240/d279802f-2030-4b2c-b750-29a0c8a4455e">


### Other information:



## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
Insert a VideoPress block (upload or use an existing video). Upload a track file. See that the process works as usual.
Try uploading a malformed .vtt file, it should show an error. 
Canceling and opening the upload modal again should allow restarting the process.
Once an upload has failed, the "Save" button won't become available again unless you load a new track on the file input.

Valid .vtt:
```
WEBVTT

1
00:00:00.000 --> 00:00:03.667
(upbeat instrumental music)

2
00:00:05.020 --> 00:00:09.000
<v ->Hey there!</v>

3
00:00:09.000 --> 00:00:12.420
I'm joined here with Ron Kendig
```

Invalid .vtt
```
xxWEBVTT

1
00:00:00.000 --> 00:00:03.667
(upbeat instrumental music)

2
00:00:05.020 --> 00:00:09.000
<v ->Hey there!</v>

3
00:00:09.000 --> 00:00:12.420
I'm joined here with Ron Kendig
```
